core/headerchain.go

```diff
@@ -231,7 +231,11 @@ func (hc *HeaderChain) WriteHeaders(headers []*types.Header) (int, error) {
 			hash = header.Hash()
 		}
 		number := header.Number.Uint64()
+		old := new(big.Int).Set(newTD)
 		newTD.Add(newTD, header.Difficulty)
+		if hc.config.TerminalTotalDifficulty != nil && old.Cmp(hc.config.TerminalTotalDifficulty) < 0 && newTD.Cmp(hc.config.TerminalTotalDifficulty) >= 0 {
+			rawdb.WriteTerminalBlockHash(hc.chainDb, header.Hash())
+		}
```

Comment:
Will this function correctly in case of a reog?

E.g. Import block A, set it as terminal. Import sibling B, set it as terminal. Import A2, reorging via A. I think there's no writeheaders in that case, so we'd remain with the discarded B as the terminal block?

Reply:
Nope